rbc/external.py

```diff
@@ -38,7 +38,7 @@ def external(cls, *args):
                     f"external function name not specified for signature {signature}"
                 )
 
-            for device in [a for a in t.annotation() or [] if a in ["CPU", "GPU"]] or [
+            for device in [a.upper() for a in t.annotation() or [] if a in ["CPU", "GPU"]] or [
```

Comment:
```suggestion
            for device in [a.upper() for a in t.annotation() or [] if a.upper() in ["CPU", "GPU"]] or [
```
?

Reply:
What about introducing a second argument to external to specify the device? Similar to how it is done in the `@udtf` decorator:

```python
fn = external(sig, device='cpu') # or 'gpu' / or both?
```

Reply:
Device hint is currently specified as an annotation to a signature. So, introducing `device` to `external` requires that `external` will check if it will conflict with the `device` hint in the signature.

I agree that specifying the device in the signature is kind of a hack and we should try to move away from using this construct. So, yes, I think your suggestion makes sense.

In udtf decorator, we use `devices` API, so:
```
fn = external(sig, devices=['cpu'])
```